host/windows/hostthread.c

```diff
@@ -13,25 +13,26 @@
 **==============================================================================
 */
 
-int oe_thread_create(oe_thread* thread, void* (*func)(void*), void* arg)
+oe_result_t oe_thread_create(oe_thread* thread, void* (*func)(void*), void* arg)
```

Comment:
I'm fine with using `oe_result_t` here but that makes it weirdly inconsistent with the rest of the functions in this file declared to mimic the POSIX function signatures. 

@mikbras Should there be a task to switch them all over to `oe_result_t` or do you prefer these to remain `int`?

Reply:
I follow up with Mike. His opinion is that we should return int for those two function to be consistent. I will make them returning POSIX error code as much as possible to address Simon's original comment.

Reply:
Also per Mike's suggestion, I have combined `oe_thread` and `oe_thread_t` into just `oe_thread_t`. Previously they are defined differently which is confusing.